src/components/detail/Detail.js

```diff
@@ -11,6 +12,31 @@ const Detail = (props) => {
   const { user } = useAuthContext();
   // console.log(user);
 
+
+  // Popup Function
+  const [popUp, setPopUp] = useState(false)
+
+  const popUpRef = useRef()
+
+  useEffect(() => {
+    
+    const handle = (e) => {
+
+      // If popup exist and clicked outside of the popup
+      if (popUpRef.current && !popUpRef.current.contains(e.target)) {
+
+        // Set popup state to false (close it)
+        setPopUp(false)
+      }
+    }
+    
+    document.addEventListener("mousedown", handle)

```

Comment:
Adding event listener to document is not so good, I think. Because whenever user click on the screen, call back function will be executed even if popup is closed. If you have time, can you think other way to implement this? I think you can use onClick handler and stopPropagation().

Reply:
Hi Taichi, I am not sure but from my understanding the hook useRef will make the eventListener only active in that ref? If not I will fixed by other way.

Reply:
You will see message shown whenever you click detail page if you add concole.log inside handle function (off course, if you add it inside if statement, it won't log out messages.), but document.addEventListener("mousedown", handle) is directly inside useEffect, so handle function will be always executed when user click somewhere in detail page.

Reply:
Thankyou for the explaination Taichi.👏👏
I will modifiy it later.